web/src/upload.js

```diff
@@ -157,7 +162,11 @@ export function show_error_message(
     message = $t({defaultMessage: "An unknown error occurred."}),
     file_id = null,
 ) {
-    get_item("send_button", config).prop("disabled", false);
+    if (config.mode === "compose") {
+        compose_validate.set_upload_in_progress(false);
+    } else {
+        get_item("send_button", config).prop("disabled", false);
+    }
```

Comment:
This seems like it can't possibly work in the event that multiple files are being uploaded at the same time, since it doesn't check if other uploads are in progress. You should be able to reproduce this by slowing down the network and arranging for only the first of several requests to fail.

`hide_upload_banner` checks whether there are no other files being uploaded, so that should work; and `upload_files` doesn't really need to consider that question, since it knows that at least the current file is being uploaded.

Hmm, but the plot thickens: `show_error_message` has 3 callers, but only one of them involves a file that was already being uploaded; and the last one, in the `upload-error` handler, is just after a call to `hide_upload_banner`, which this logic duplicates. So the fix might be to just delete this line here rather than adding the conditional.

(This is presumably a preexisting bug; feel free to just fix it in its own prep or follow-up commit)

Reply:
Thanks for catching this! I moved down the code for disabling send on upload inside the for loop, right before the code adding the "Uploading ..." banner for each file, and removed the send enabling code from `show_error_message`. 

This ensures that the send disabling and re-enabling is in sync with the addition and removal of the "Uploading ..." banner, and fixes the multiple upload 1 failed bug!